src/lib.rs

```diff
@@ -462,6 +465,48 @@ pub mod channel {
                     Sender::Boxed(f) => f(value).await,
                 }
             }
+
+            /// Applies a filter before sending.
+            ///
+            /// Messages that don't pass the filter are dropped.
+            pub fn with_filter(self, f: impl Fn(&T) -> bool + Send + Sync + 'static) -> Sender<T>
```

Comment:
not super convinced that these are useful on the oneshot sender? but maybe they are :)

Reply:
🤷 just did it for completeness.